main.c

```diff
@@ -1196,25 +1236,17 @@ int main(int argc, char **argv) {
 		return 2;
 	}
 
-	if (state.zxdg_output_manager) {
-		struct swaylock_surface *surface;
-		wl_list_for_each(surface, &state.surfaces, link) {
-			surface->xdg_output = zxdg_output_manager_v1_get_xdg_output(
-						state.zxdg_output_manager, surface->output);
-			zxdg_output_v1_add_listener(
-					surface->xdg_output, &_xdg_output_listener, surface);
-		}
-		wl_display_roundtrip(state.display);
-	} else {
-		swaylock_log(LOG_INFO, "Compositor does not support zxdg output "
-				"manager, images assigned to named outputs will not work");
-	}
-
 	struct swaylock_surface *surface;
 	wl_list_for_each(surface, &state.surfaces, link) {
 		create_layer_surface(surface);
 	}
 
+	wl_list_for_each(surface, &state.surfaces, link) {
+		while (surface->ready > 0) {
+			wl_display_roundtrip(state.display);
```

Comment:
dispatch should be enough here, no need to roundtrip. Also please check for errors so that we don't enter in a busy-loop if the connection is broken (e.g. because of a protocol error or the compositor dying).